app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/WifiDirectBroadcastReceiver.java

```diff
@@ -0,0 +1,81 @@
+package org.kiwix.kiwixmobile.zim_manager.local_file_transfer;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.net.NetworkInfo;
+import android.net.wifi.p2p.WifiP2pDevice;
+import android.net.wifi.p2p.WifiP2pManager;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import org.kiwix.kiwixmobile.R;
+
+import static org.kiwix.kiwixmobile.zim_manager.local_file_transfer.LocalFileTransferActivity.showToast;
+
+/**
+ * Helper class for the local file sharing module.
+ *
+ * Handles the broadcasts pertaining to the wifi p2p group formed in WiFi Direct. Works along with
+ * the wifi p2p manager in {@link LocalFileTransferActivity}.
+ */
+public class WifiDirectBroadcastReceiver extends BroadcastReceiver {
+
+  private WifiP2pManager manager;
+  private WifiP2pManager.Channel channel;
+  private LocalFileTransferActivity wifiActivity;
+
+  public WifiDirectBroadcastReceiver(@NonNull WifiP2pManager manager, @NonNull WifiP2pManager.Channel channel,
+      @NonNull LocalFileTransferActivity activity) {
+    super();
+    this.manager = manager;
+    this.channel = channel;
+    this.wifiActivity = activity;
+  }
+
+  @Override
+  public void onReceive(@NonNull Context context, @NonNull Intent intent) {
+    String action = intent.getAction();
+
+    if (WifiP2pManager.WIFI_P2P_STATE_CHANGED_ACTION.equals(action)) {
+      // Update wifi p2p state
+      int wifiP2pState = intent.getIntExtra(WifiP2pManager.EXTRA_WIFI_STATE, -1);
+
+      if (wifiP2pState == WifiP2pManager.WIFI_P2P_STATE_ENABLED) {
```

Comment:
In this case it would be
```
listener.onWifiP2PStateChanged(wifiP2pState == WifiP2pManager.WIFI_P2P_STATE_ENABLED)
```
and on the impl side 

```
@Override
public void onWifiP2PStateChanged(boolean isEnabled){
setWifiP2pEnabled(isEnabled);
if(!isEnabled){
listener.onWifiP2PDisabled();
}
}
```

Reply:
Does [this](https://github.com/kiwix/kiwix-android/pull/1276/commits/ba00ece2ca1b9d3cba9ee40eba1c0a2b2022c8fd#diff-7e01f6f425daaca67175b74c839c6af2R135) work as well?

Reply:
I would say no. You are leaking information from the BroadcastReceiver. WDM doesn't need to know about these WifiP2PManager constants as the complexity was dealt with inside the BroadcastReceiver, there is no way to know here in WDM what this int means without looking at another class.
```
onWifiP2PStateChanged(boolean isEnabled){
```
tells you everything you want to know about the p2p state in this class. If you ever need to extend the number of values this returns then your own enum would be even better

Reply:
Based on what I understand, in this [subsequent review](https://github.com/kiwix/kiwix-android/pull/1276#discussion_r311934064) as well, what you're saying is that WDM should get exactly what it needs from the BroadcastReceiver and nothing more?

Reply:
Yup, I'll look forward to the commit

Reply:
Does [this](https://github.com/kiwix/kiwix-android/pull/1276/commits/1baafc9db4270b2ccf492be740c443791b01ae2b#diff-e84e03042026401a0660fa610753ae45R58) work?

Reply:
Yup, great